utils/generate_fmtargs.py

```diff
@@ -0,0 +1,43 @@
+MAX_ARGS = 64
+
+print('#define __ARG_N(' + ', '.join(['_' + str(i) for i in range(1, MAX_ARGS, 1)]) + ', N, ...) N')
+
+print('\n#define __RSEQ_N() ' + ', '.join([str(i) for i in range(MAX_ARGS - 1, -1, -1)]))
+
+print('\n#define COMPACT_FMT_2(fmt, value) fmt')
+for i in range(4, MAX_ARGS + 1, 2):
+    print('#define COMPACT_FMT_{}(fmt, value, ...) fmt COMPACT_FMT_{}(__VA_ARGS__)'.format(i, i - 2))
+
+
+print('\n#define COMPACT_VALUES_2(fmt, value) value')
+for i in range(4, MAX_ARGS + 1, 2):
+    print('#define COMPACT_VALUES_{}(fmt, value, ...) value, COMPACT_VALUES_{}(__VA_ARGS__)'.format(i, i - 2))
+
+
+# #define __ARG_N( \
```

Comment:
For those curious, this was just my reference implementation I was comparing against.